ruleset.xml

```diff
@@ -1,5 +1,137 @@
+    <rule ref="Squiz.Commenting.DocCommentAlignment" />
+
+    <!-- Control structures -->
+    <rule ref="Generic.ControlStructures.InlineControlStructure" />
+    <rule ref="Squiz.ControlStructures.ControlSignature" />
+    <rule ref="../TYPO3SniffPool/Sniffs/ControlStructures/DisallowElseIfConstructSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/ControlStructures/AlignedBreakStatementSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/ControlStructures/DisallowEachInLoopConditionSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/ControlStructures/UnusedVariableInForEachLoopSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/ControlStructures/ExtraBracesByAssignmentInLoopSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/ControlStructures/SwitchDeclarationSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/ControlStructures/TernaryConditionalOperatorSniff.php" />
+
+    <!-- Debug -->
+    <rule ref="../TYPO3SniffPool/Sniffs/Debug/DebugCodeSniff.php">
+        <exclude-pattern>*typo3/sysext/core/Classes/Utility/DebugUtility.php</exclude-pattern>
+    </rule>
+
+    <!-- Files -->
+    <rule ref="Generic.Files.OneClassPerFile">
+        <exclude-pattern>*typo3/sysext/core/Migrations/Code/LegacyClassesForIde.php</exclude-pattern>
+    </rule>
+    <rule ref="Generic.Files.OneInterfacePerFile" />
+    <rule ref="Generic.Files.ByteOrderMark" />
+    <rule ref="Generic.Files.LineEndings" />
+    <rule ref="Generic.Files.LineLength">
+        <properties>
+            <property name="lineLimit" value="130"/>
+            <property name="absoluteLineLimit" value="200"/>
+        </properties>
+    </rule>
+    <rule ref="../TYPO3SniffPool/Sniffs/Files/IncludingFileSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/Files/FilenameSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/Files/FileExtensionSniff.php" />
+    <rule ref="Zend.Files.ClosingTag" />
+
+    <!-- Formatting -->
+    <rule ref="Generic.Formatting.DisallowMultipleStatements" />
+
+    <!-- Functions -->
+    <rule ref="Generic.Functions.FunctionCallArgumentSpacing" />
+    <rule ref="Generic.Functions.OpeningFunctionBraceKernighanRitchie" />
+
+    <!-- Loops -->
+
+    <!-- NamingConventions -->
+    <rule ref="../TYPO3SniffPool/Sniffs/NamingConventions/ValidVariableNameSniff.php" />
+    <rule ref="Generic.NamingConventions.ConstructorName" />
+    <rule ref="../TYPO3SniffPool/Sniffs/NamingConventions/ValidFunctionNameSniff.php" />
+
+    <!-- Objects -->
+
+    <!-- Operators -->
+
+    <!-- PHP -->
+    <rule ref="Generic.PHP.CharacterBeforePHPOpeningTag" />
+    <rule ref="Generic.PHP.DisallowShortOpenTag" />
+    <rule ref="Squiz.PHP.NonExecutableCode" />
+    <rule ref="../TYPO3SniffPool/Sniffs/PHP/DisallowMultiplePHPTagsSniff.php" />
+    <rule ref="Generic.PHP.UpperCaseConstant" />
+    <rule ref="Generic.PHP.DeprecatedFunctions" />
+    <rule ref="Squiz.PHP.Eval">
+        <exclude-pattern>*/tests/*</exclude-pattern>
```

Comment:
We should not need any exclude here, as we should only lint our own files, see comment above.